### ISSUE TITLE
avoid `modernize-avoid-c-arrays` with `main` style parameters

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#60411]
```cpp
extern int main_impl(int, char *[]);

int main(int argc, char *argv[])
{
    return main_impl(argc, argv);
}
```
```console
<source>:1:27: warning: do not declare C-style arrays, use std::array<> instead [modernize-avoid-c-arrays]
extern int main_impl(int, char *[]);
                          ^
```
https://godbolt.org/z/h7PT3aKnE

The check is obviously there to avoid continuous usage of those unsafe parameters but if you implement the handling into `main()` directly that would still be the case.

But you cannot simply convert it a `std::array` and need some template function to generate that which each application would have to implement by itself. And even then you need to suppress the warning on the conversion function.

So I think it would make sense to be able to avoid the warning on those exact parameters. It is probably a common pattern (Boost and googletest offer custom `main` implementations IIRC).

You could also make this optional via a "do not warn on main-like" option which is disabled by default to enforce better safety out-of-the-box.

**Comment by PiotrZSL** [comment]
Check already ignores simple main(), only thing thats missing is configuration for other functions. to be ignored.

**Comment by carlosgalvezp** [comment]
Could you simple use char ** argv?

**Comment by vbvictor** [comment]
This could be done with NOLINT or char**
